src/test/javascript/spec/component/overview/course-discussion/course-discussion.component.spec.ts

```diff
@@ -0,0 +1,373 @@
+        expect(component.currentPostContextFilter).to.be.deep.equal({
+            courseId: metisCourse.id,
+            courseWideContext: undefined,
+            exerciseId: undefined,
+            lectureId: undefined,
+        });
+    }));
+
+    it('should initialize formGroup correctly', fakeAsync(() => {
+        component.ngOnInit();
+        tick();
+        expect(component.formGroup.get('context')?.value).to.be.deep.equal({
+            courseId: metisCourse.id,
+            courseWideContext: undefined,
+            exerciseId: undefined,
+            lectureId: undefined,
+        });
+        expect(component.formGroup.get('sortBy')?.value).to.be.equal(PostSortCriterion.CREATION_DATE);
+        expect(component.formGroup.get('sortDirection')?.value).to.be.equal(SortDirection.DESC);
+    }));
+
+    it('should initialize overview page with course posts for default settings correctly', fakeAsync(() => {
+        component.ngOnInit();
+        tick();
+        expect(component.formGroup.get('context')?.value).to.be.deep.equal({
+            courseId: metisCourse.id,
+            courseWideContext: undefined,
+            exerciseId: undefined,
+            lectureId: undefined,
+        });
+        expect(component.formGroup.get('sortBy')?.value).to.be.equal(PostSortCriterion.CREATION_DATE);
+        expect(component.formGroup.get('sortDirection')?.value).to.be.equal(SortDirection.DESC);
+        fixture.detectChanges();
+        const searchInput = getElement(fixture.debugElement, 'input[name=searchText]');
+        expect(searchInput.textContent).to.be.equal('');
+        const contextOptions = getElement(fixture.debugElement, 'select[name=context]');
+        // select should provide all context options
+        expect(contextOptions.textContent).contains(metisCourse.title);
+        expect(contextOptions.textContent).contains(metisLecture.title);
+        expect(contextOptions.textContent).contains(metisExercise.title);
+        // course should be selected
+        const selectedContextOption = getElement(fixture.debugElement, 'select[name=context]');
+        expect(selectedContextOption.value).contains(metisCourse.title);
+        // creation date should be selected as sort criterion
+        const selectedSortByOption = getElement(fixture.debugElement, 'select[name=sortBy]');
+        expect(selectedSortByOption.value).to.exist;
+        // descending should be selected as sort direction
+        const selectedDirectionOption = getElement(fixture.debugElement, 'select[name=sortDirection]');
+        expect(selectedDirectionOption.value).to.exist;
+        // show correct number of posts found
+        const postCountInformation = getElement(fixture.debugElement, '.post-result-information');
+        expect(postCountInformation.innerHTML).to.not.be.empty;
+    }));
+
+    it('should invoke metis service with foreReload false when search text changed', fakeAsync(() => {
```

Comment:
```suggestion
    it('should invoke metis service with forReload false when search text changed', fakeAsync(() => {
```
Or perhaps
```suggestion
    it('should invoke metis service with beforeReload false when search text changed', fakeAsync(() => {
```
Not sure

Reply:
it would have been forceReload, I changed the description